sdks/go/pkg/beam/core/graph/window/strategy.go

```diff
@@ -16,12 +16,16 @@
 // Package window contains window representation, windowing strategies and utilities.
 package window
 
+import (
+	pipepb "github.com/apache/beam/sdks/go/pkg/beam/model/pipeline_v1"
+)
+
 // WindowingStrategy defines the types of windowing used in a pipeline and contains
 // the data to support executing a windowing strategy.
 type WindowingStrategy struct {
 	Fn *Fn
-
 	// TODO(BEAM-3304): trigger support
+	Trigger *pipepb.Trigger
```

Comment:
As a rule, we avoid having users have access to, or deal with, the proto directly, including via any of the built up structs. The protos are an implementation detail, not an interface for users to use.

This largely means building up SDK specific mechanisms that are then translated to the proto in graphx/translate.go. Essentially, the graph/** directories shouldn't be depending on the protos.

That said, this is fine during development as Pane plumbing and such is being developed.